Plugins~/Src/MeshSyncClientBlender/msblenGeometryNodeUtils.h

```diff
@@ -40,10 +40,10 @@ class GeometryNodesUtils
     void each_instanced_object(std::function<void(Object*, Object*, SharedVector<mu::float4x4>, bool)> handler);
 
     /// <summary>
-    /// /// Converts the world matrix from blender to Unity coordinate system
-    /// /// </summary>
-    mu::float4x4 blenderToUnityWorldMatrix(const mu::float4x4& blenderMatrix);
-
+    /// Converts the world matrix from blender to Unity coordinate system
+    /// </summary>
+    mu::float4x4 blenderToUnityWorldMatrix(const Object* obj, const mu::float4x4& blenderMatrix) const;
```

Comment:
That would not work with the `applyCorrectionIfNeeded` method. That method uses `is_light` from the mesh utils and doesn't check the type directly.

Reply:
Both the is_camera and is_light are a type check
```
bool is_camera(const Object *obj) { return obj->type == OB_CAMERA; }
bool is_light(const Object *obj) { return obj->type == OB_LAMP; }
```

can we not just pass that short? And the method can just check the parameter instead of using the utility

Reply:
you can create an overload that takes the enum instead of the object. Then the function that takes object can be simplified to
```
applyCorrectionIfNeeded(obj, ...){
applyCorrectionIfNeeded(obj->type, ..)
}
```

Reply:
Yes but `applyCorrectionIfNeeded` specifically uses `is_camera` and `is_light` to avoid direct type checks. Unless we don't care about that abstraction anymore and want to get rid of those methods.
I think it would better to just pass the object instead of removing the abstraction just so we can pass a type from the object with no real benefit.

Reply:
I have 2 longer term concerns: 
One is passing around raw pointers to blender objects. This is probably necessary for a lot of our code, but if there is another way, it might be good to start thinking about how we can change our code.

The other is, the function takes an object to convert a matrix. I understand why, but it seems to me that this function should do what it describes, and the object parameter looks strange. 

In any case, I approve the PR as this is not blocking